src/arm/debugger/memory-debugger.c

```diff
@@ -91,13 +91,13 @@ CREATE_MULTIPLE_WATCHPOINT_SHIM(loadMultiple, WATCHPOINT_READ)
 CREATE_MULTIPLE_WATCHPOINT_SHIM(storeMultiple, WATCHPOINT_WRITE)
 CREATE_SHIM(setActiveRegion, void, (struct ARMCore* cpu, uint32_t address), address)
 
-static bool _checkWatchpoints(struct ARMDebugger* debugger, uint32_t address, struct mDebuggerEntryInfo* info, enum mWatchpointType type, uint32_t newValue, int width) {
-	--width;
+static bool _checkWatchpoints(struct ARMDebugger* debugger, uint32_t minAddress, struct mDebuggerEntryInfo* info, enum mWatchpointType type, uint32_t newValue, int width) {
```

Comment:
I'm not really fond of renaming `address` to `minAddress`, especially since it's not even accurate once you factor in alignment. GBA doesn't have misaligned loads, so addressing a 32-bit value at 0x2 will start access at 0x0 and end at 0x3, not start at 0x2 and end at 0x5. Having a `minAddress` is fine, but it's not equivalent to `address` with misaligned loads. Not to mention the fact that it causes unnecessary and confusing code churn later in the function.

Reply:
Yes, at least in the case of the GBA. But we don't care about platforms that support unaligned loads here.